group/assignment6/RearrangingCars.java

```diff
@@ -0,0 +1,101 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+class Move{
+    private int car;
+    private String old_space;
+    private String new_space;
+
+    public Move(int car, String old_space, String new_space) {
+        this.car = car;
+        this.old_space = old_space;
+        this.new_space = new_space;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        Move move = (Move) obj;
+        return (move.getCar() == car && move.getOld_space() == old_space && move.getNew_space() == new_space);
+    }
+
+    public int getCar() {
+        return car;
+    }
+
+    public String getOld_space() {
+        return old_space;
+    }
+
+    public String getNew_space() {
+        return new_space;
+    }
+}
+
+public class RearrangingCars {
+
+    private Map<String, Integer> start_state;
```

Comment:
I wanted to use String because I thought of cases with a lot of data and the parking spaces can be named "A", "B", ..., "AA", "AB", ... I can change it to Character if you think it's better that way :)

Reply:
Oh I didn't thought about that case ;) So I think you shouldn't change this.